test/e2e/sagemaker/tests/test_transformjob.py

```diff
@@ -0,0 +1,154 @@
+        assert 'ackResourceMetadata' in resource['status'] and \
+            'arn' in resource['status']['ackResourceMetadata']
+        return resource['status']['ackResourceMetadata']['arn']
+
+    def _get_sagemaker_transformjob_arn(self, sagemaker_client, transformjob_name: str):
+        try:
+            transformjob = sagemaker_client.describe_transform_job(TransformJobName=transformjob_name)
+            return transformjob['TransformJobArn']
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find a transformJob with the name {transformjob_name}")
+            return None
+
+    def _get_sagemaker_transformjob_status(self, sagemaker_client, transformjob_name: str):
+        try:
+            transformjob = sagemaker_client.describe_transform_job(TransformJobName=transformjob_name)
+            return transformjob['TransformJobStatus']
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find a transformJob with the name {transformjob_name}")
+            return None
+
+    def test_create_transformjob(self, xgboost_transformjob):
+        (reference, resource) = xgboost_transformjob
+        assert k8s.get_resource_exists(reference)
+
+    def test_transformjob_has_correct_arn(self, sagemaker_client, xgboost_transformjob):
+        (reference, _) = xgboost_transformjob
+        resource = k8s.get_resource(reference)
+        transformjob_name = resource['spec'].get('transformJobName', None)
+
+        assert transformjob_name is not None
+
+        resource_transformjob_arn = self._get_resource_transformjob_arn(resource)
+        expected_transformjob_arn = self._get_sagemaker_transformjob_arn(
+            sagemaker_client, transformjob_name)
+
+        assert resource_transformjob_arn == expected_transformjob_arn
+
+    def test_transformjob_has_created_status(self, sagemaker_client, xgboost_transformjob):
+        (reference, _) = xgboost_transformjob
+        resource = k8s.get_resource(reference)
+        transformjob_name = resource['spec'].get('transformJobName', None)
+
+        assert transformjob_name is not None
+
+        current_transformjob_status = self._get_sagemaker_transformjob_status(
+            sagemaker_client, transformjob_name)
+        expected_transformjob_status_list = self._get_created_transformjob_status_list()
+        assert current_transformjob_status in expected_transformjob_status_list
+        
+
+    def test_transformjob_has_stopped_status(self, sagemaker_client, xgboost_transformjob):
+        (reference, _) = xgboost_transformjob
+        resource = k8s.get_resource(reference)
+        transformjob_name = resource['spec'].get('transformJobName', None)
+
+        assert transformjob_name is not None
+
+        # Delete the k8s resource.
+        k8s.delete_custom_resource(reference)
+        # TODO: This sleep could be replaced by a wait loop but this is sufficient for now.
+        time.sleep(5)
```

Comment:
`delete_custom_resource` seems to wait until the custom resource is deleted from the API server before it returns. @surajkota Are these sleeps still necessary? I understand ACK does not wait for the status to change to `Stopping`, but surely once SageMaker has received the `StopTransformJob` API call, it changes the state to `Stopping` instantly?

Reply:
right, dont need sleep now. Check other test files on latest commit

Reply:
Fixed.